slsim/roman_image_simulation.py

```diff
@@ -315,3 +320,123 @@ def _get_wcs_dict(ra, dec, date):
+    :type psf_directory: string
+    :param t_obs: an observation time [day]. This is applicable only for
+        variable source. In case of point source, if we do not provide
+        t_obs, considers no variability in the lens.
+    :param with_source: If True, simulates image with extended source in
+        lens configuration.
+    :param with_deflector: If True, simulates image with deflector.
+    :param ra: Coordinate in space used to generate sky background
+    :type ra: float between 15 and 45
+    :param dec: Coordinate in space used to generate sky background
+    :type dec: float between -45 and -15
+    :param date: Date used to generate sky background
+    :type date: datetime.datetime class
+    :param add_noise: determines whether sky background and detector effects are added or not
+    :type add_background: bool
+    :param seed: An rng seed used for generating detector effects in galsim
+    :type seed: integer or None
+    :return: lens image in roman filter
+    """
+    delta_pix = transformmatrix_to_pixelscale(transform_pix2angle)
+    psf_interp = get_psf(
+        band=band,
+        detector=detector,
+        detector_pos=detector_pos,
+        oversample=oversample,
+        psf_directory=psf_directory,
+    )
+    psf_kernel = psf_interp.image.array
+    deflector_image = sharp_image(
+        lens_class=lens_class,
+        band=band,
+        mag_zero_point=mag_zero_point,
+        delta_pix=delta_pix,
+        num_pix=num_pix,
+        with_source=with_source,
+        with_deflector=with_deflector,
+    )
+    convolved_deflector = convolved_image(deflector_image, psf_kernel=psf_kernel)
+    ps_image = point_source_image_at_time(
+        lens_class,
+        band=band,
+        mag_zero_point=mag_zero_point,
+        delta_pix=0.11,
+        num_pix=num_pix,
+        psf_kernel=psf_kernel,
+        transform_pix2angle=transform_pix2angle,
+        time=t_obs,
+    )
+    ps_image = np.nan_to_num(ps_image, nan=0)
+    image = convolved_deflector + ps_image
+    image_galsim = galsim.ImageF(image, scale=delta_pix)
+    noise_galsim = galsim.ImageF(num_pix, num_pix, scale=delta_pix)
+    kwargs_single_band = image_quality_lenstronomy.kwargs_single_band(
+        observatory="Roman", band=band
+    )
+    _exposure_time = kwargs_single_band["exposure_time"]
+    if add_noise is True:
+        # Obtain sky background corresponding to certain band and add it to the image
+        # Requires webbpsf data files to use
+        noise_galsim = add_roman_background(
+            noise_galsim, band, detector, num_pix, _exposure_time, ra, dec, date
+        )
+
+        # Add detector effects and get the resulting array
+        rng = galsim.UniformDeviate(seed)
+        roman.allDetectorEffects(
+            noise_galsim, prev_exposures=(), rng=rng, exptime=_exposure_time
+        )
+
+    image_array = image_galsim.array
+    noise_array = noise_galsim.array / _exposure_time
+    final_image = image_array + noise_array
```

Comment:
checking: this is only Gaussian noise. Is there any chance to get the Poisson noise in here?